waiter/test/waiter/scheduler/kubernetes_test.clj

```diff
@@ -129,9 +131,32 @@
                                           {:service-id->service-description-fn (constantly service-description)})
           replicaset-spec ((:replicaset-spec-builder-fn scheduler) scheduler "test-service-id" service-description)]
       (is (= {:waiter/port-count "3"
-              :waiter/service-id "test-service-id"}
+              :waiter/service-id "test-service-id"
+              :waiter/service-port 8330}
              (get-in replicaset-spec [:spec :template :metadata :annotations]))))))
 
+(deftest replicaset-spec-with-reverse-proxy
+  (with-redefs [config/retrieve-cluster-name (constantly "test-cluster")
+                config/retrieve-waiter-principal (constantly "waiter@test.com")]
+    (let [scheduler (make-dummy-scheduler ["test-service-id"])
+          replicaset-spec ((:replicaset-spec-builder-fn scheduler) scheduler "test-service-id"
+                           (assoc dummy-service-description "env" {(:reverse-proxy-flag (:proxy-options scheduler))
+                                                                   "yes"}))]
+
+      (testing "replicaset has waiter/service-port annotation"
+        (is (contains? (get-in replicaset-spec [:spec :template :metadata :annotations]) :waiter/service-port)))
+
+      (testing "service-port and waiter ports are correct"
+        (let [service-port (-> "test-service-id" hash (mod 100) (* 10) (+ (:pod-base-port scheduler)))
+              port0 (+ service-port 1)]
+          (is (= service-port (get-in replicaset-spec [:spec :template :metadata :annotations :waiter/service-port])))
+          (is (= port0 (get-in replicaset-spec [:spec :template :spec :containers 0 :ports 0 :containerPort])))))
+
+      (testing "waiter/port-count annotation is correct"
+        (let [port-count (inc (get service-description "ports"))]
```

Comment:
A map based off of `dummy-service-description` was used as the input. We should extract that map into a variable and use it as the input here.

Reply:
Done